src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,167 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as os from 'os';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as mssql from 'vscode-mssql';
+import { defaultBindingResult, defaultSqlBindingTextLines, sqlBindingResult } from '../constants/constants';
+import * as LocalizedConstants from '../constants/localizedConstants';
+import { AzureFunctionsService } from '../services/azureFunctionsService';
+import * as azureFunctionUtils from '../azureFunction/azureFunctionUtils';
+import * as constants from '../constants/constants';
+import * as path from 'path';
+import { executeCommand } from '../utils/utils';
+const sqlBindingNugetSource = 'https://www.myget.org/F/azure-appservice/api/v3/index.json';
+let selectedProjectFile: string | undefined = '';
+export class AzureFunctionProjectService {
+
+	constructor(private azureFunctionsService: AzureFunctionsService) {
+	}
+
+	public async createAzureFunction(connectionString: string, schema: string, table: string): Promise<void> {
+		const azureFunctionApi = await azureFunctionUtils.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			return;
+		}
+		if (!await isAzureFunctionProjectOpen()) {
+			vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened);
+			return;
+		}
+
+		// because of an AF extension API issue, we have to get the newly created file by adding
+		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
+		const newFilePromise = getNewFunctionFile();
+
+		// get function name from user
+		const functionName = await vscode.window.showInputBox({
+			title: constants.functionNameTitle,
+			value: table
+		});
+
+		// create C# HttpTrigger
+		await azureFunctionApi.createFunction({
+			language: 'C#',
+			templateId: 'HttpTrigger',
+			functionName: functionName
+		});
+
+		// once steps are addressed we add SQL Bindings Nuget and Connection String to local.settings.json
+		// and then create a new functions file
+		await addNugetReferenceToProjectFile();
+		await addConnectionStringToConfig(connectionString);
+		const functionFile = await newFilePromise;
+
+		let objectName = escapeObjectName(schema, table);
+		await this.azureFunctionsService.addSqlBinding(
+			mssql.BindingType.input,
```

Comment:
Since this is hardcoded to input binding, we need to improve the strings for the experience.

@dzsquared can you help with the string for the context menu? Should it say "input binding"?

Reply:
We definitely should - it'd be nice to support creating an output binding as well but until that's done we should be explicit as possible

Reply:
"Create Azure Function with SQL input binding"